src/app/pages/signing/signing.component.ts

```diff
@@ -78,13 +80,16 @@ export class SigningComponent implements OnInit {
   /**
    * Extract query parameters from ActivatedRoute and maximum message length from MsgIoService
    */
-  constructor(private route: ActivatedRoute, private msgIo: MsgIoService, private router: Router) {
+  constructor(private route: ActivatedRoute, private authService: AuthService, private msgIo: MsgIoService, private router: Router) {
     this.route.queryParams.subscribe(params => {
-      this.uid = params['uid'];
-      this.key = params['key'];
+      this.uid = params.uid;
+      this.key = params.key;
     });
 
     this.maxCharCount = MsgIoService.maxMessageLength;
+
+    authService.displayName(this.uid).then(userName => this.name = userName).catch(error => alert(error));
```

Comment:
Avoid using the constructor for initializing variables. All of these operations (Lines 84-91) should be moved to `ngOnInit()`.